peps/pep-0694.rst

```diff
@@ -832,6 +757,72 @@ Besides the standard ``meta`` key, this has the following top level keys:
 The ``message`` and ``source`` strings do not have any specific meaning, and are intended for human
 interpretation to aid in diagnosing underlying issue.
 
+File Upload Mechanisms
+----------------------
+
+Servers **MUST** implement :ref:`required file upload mechansisms <required-file-upload-mechanisms>`.
+Such mechanisms serve as a fallback if no server specific implementations exist.
+
+Each major version of the Upload API **MUST** specify at least one required File Upload Mechanism.
+
+New required mechanisms **MUST NOT** be added and existing required mechanisms **MUST NOT** be removed
+without an update to the :ref:`major version <versioning>`.
+
+A given server **MAY** implement an arbitrary number of server specific mechanisms
+and is responsible for documenting their usage.
+Server specific implementations **MUST** be prefixed with ``vnd-``
```

Comment:
Ah, now that I'm here (and admittedly, running out of gas -- it's EOW) I see that my earlier comment about the mechanism name of `pypi-atomic` wouldn't be appropriate because it would be a required protocol and not a vendor protocol.  So maybe that would be changed to `atomic` instead.